pkg/registry/csv.go

```diff
@@ -412,3 +414,60 @@ func (csv *ClusterServiceVersion) GetSubstitutesFor() string {
 	}
 	return substitutesFor
 }
+
+func (csv *ClusterServiceVersion) UnmarshalJSON(data []byte) error {
+
+	if err := csv.UnmarshalSpec(data); err != nil {
+		return err
+	}
+	if err := csv.UnmarshalTypeMeta(data); err != nil {
+		return err
+	}
+	if err := csv.UnmarshalObjectMeta(data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (csv *ClusterServiceVersion) UnmarshalSpec(data []byte) error {
+	var m map[string]json.RawMessage
+	if err := json.Unmarshal(data, &m); err != nil {
+		return errors.New(prettyunmarshaler.NewJSONUnmarshalError(data, err).Pretty())
+	}
+
+	specRaw, ok := m["spec"]
+	if !ok {
+		return nil
+	}
+	var spec json.RawMessage
+	if err := json.Unmarshal(specRaw, &spec); err != nil {
+		return errors.New(prettyunmarshaler.NewJSONUnmarshalError(specRaw, err).Pretty())
+	}
```

Comment:
Do we need this unmarshal if the first unmarshal does a `map[string]json.RawMessage`? `specRaw` would already be a `json.RawMessage` here so could we just set `csv.Spec = specRaw` and it work the same?